src/core/annotation.js

```diff
@@ -80,8 +84,9 @@ AnnotationFactory.prototype = /** @lends AnnotationFactory.prototype */ {
     var parameters = {
       xref: xref,
       dict: dict,
-      ref: ref,
+      ref: isRef(ref) ? ref : null,
```

Comment:
@timvandermeij Please note that this change isn't necessary in order to fix #7569, but when looking through this file I noticed that there's currently one code-path where we're attempting to compare something to the annotation `ref` (see https://github.com/mozilla/pdf.js/blob/master/src/core/annotation.js#L627-L644).
Hence I wanted to prevent a future issue by only passing in the `ref` when it's actually valid.